learning_journal/learning_journal/views.py

```diff
@@ -1,47 +1,31 @@
-from pyramid.response import Response
 from pyramid.view import view_config
-import transaction
 import datetime
-from jinja2 import Markup
-import markdown
-from sqlalchemy import update
 from .forms import EntryCreateForm, EntryUpdateForm
 from pyramid.httpexceptions import HTTPFound
 
-
-from sqlalchemy.exc import DBAPIError, ResourceClosedError, IntegrityError
-
 from .models import (
     DBSession,
     Entry,
 )
 
 
-# @view_config(route_name='list', renderer='string')
 @view_config(route_name='list', renderer='templates/pretty.jinja2')
 def list_view(request):
-    display = DBSession().query(Entry.metadata.tables['entries']).all()[::-1][:10]
+    display = DBSession.query(Entry).order_by(Entry.created)
     return {"content": display, "header": "entries.order_by(latest)[:10]"}
 
 
 @view_config(route_name='detail', renderer='templates/detail.jinja2')
 def detail_view(request):
     id_ = request.matchdict.get('entry_id')
-    display = DBSession().query(Entry.metadata.tables['entries']).filter_by(id=id_).one()
-    md = markdown.Markdown(safe_mode='replace', html_replacement_text='--RAW HTML NOT ALLOWED--')
-    message = md.convert(display[2])
-
-    return {"message": message, "header": display[0], "title": display[1], "time": display[3]}
+    entry = DBSession().query(Entry).get(id_)
+    return {'entry': entry}
```

Comment:
What if there is no entry with that id?  What does your user see?  What *should* your user see?